prow/pod-utils/decorate/podspec.go

```diff
@@ -76,7 +76,7 @@ func LabelsAndAnnotationsForSpec(spec kube.ProwJobSpec, extraLabels, extraAnnota
 	jobNameForLabel := spec.Job
 	if len(jobNameForLabel) > validation.LabelValueMaxLength {
 		// TODO(fejta): consider truncating middle rather than end.
-		jobNameForLabel = strings.TrimRight(spec.Job[:validation.LabelValueMaxLength], "-")
+		jobNameForLabel = strings.TrimRight(spec.Job[:validation.LabelValueMaxLength], ".-")
```

Comment:
maybe we should not try to stick these values in labels? /shrug

Reply:
Maybe. Anecdotally it's really useful to be able to `kubectl get ... --selector` for job names, even snippets are usually uniquely identifying.